src/halmos/__main__.py

```diff
@@ -733,14 +734,20 @@ def future_callback(future_model):
             f"# of potential paths involving assertion violations: {len(future_models)} / {len(result_exs)}  (--solver-threads {args.solver_threads})"
         )
 
-    if args.early_exit:
-        while not (
-            len(counterexamples) > 0 or all([fm.done() for fm in future_models])
-        ):
+    # display assertion solving progress
+    with Status("solving:") as status:
+        while True:
+            if args.early_exit and len(counterexamples) > 0:
+                break
+            done = sum(fm.done() for fm in future_models)
+            total = len(future_models)
+            if done == total:
+                break
+            status.update(f"solving: {done} / {total}")
             time.sleep(1)
```

Comment:
downside of this polling loop:  the assertion solving phase takes a minimum of 1s, even if the solver finishes in less than 1s.  wouldn't be a big deal in most cases, but this causes the regression test set to take longer, because it involves many small tests.